Causing build error: "Module not found: Default condition should be last one"

When trying to use this library in a next.js project I get a build error with the message "Module not found: Default condition should be last one"

According to https://stackoverflow.com/questions/76127288/module-not-found-error-default-condition-should-be-last-one this is cause by the "default" key in an export map not being the literally last key, which kinda sounds absurd but I'm not too familiar with how export maps work.

https://nodejs.org/api/packages.html indicates that the "default" key "should always come last", but maybe it actually must?

Confirmed that swapping the order of the keys in the package.json file within my node_modules fixes the issue

--- a/package.json
+++ b/package.json
@@ -9,20 +9,20 @@
   "exports": {
     ".": {
       "import": {
-        "default": "./lib/index.min.js",
-        "types": "./lib/index.d.ts"
+        "types": "./lib/index.d.ts",
+        "default": "./lib/index.min.js"
       },
       "types": {
         "import": {
-          "default": "./lib/index.d.ts",
-          "types": "./lib/index.d.ts"
+          "types": "./lib/index.d.ts",
+          "default": "./lib/index.d.ts"
         }
       }
     },
     "./full": {
       "import": {
-        "default": "./lib/index.js",
-        "types": "./lib/index.d.ts"
+        "types": "./lib/index.d.ts",
+        "default": "./lib/index.js"
       }
     },
     "./package.json": "./package.json"

